decontam-Score-viz update

jordenrabasco opened the pull request:
This pull request includes a variety of new features including an updated look, a sortable decontam score table with associated sequences, and downloadable fasta files of the contaminant and non-contaminant features.

jordenrabasco committed:
initial integration

-working integration of decontam and quality control

jordenrabasco committed:
clean plugin_steup.py

-removes redundant code and updates look of code

jordenrabasco committed:
Clean setup.py

-improves spacing

jordenrabasco committed:
clean transfomer.py

-removes old code and updates spacing

jordenrabasco committed:
Clean test_stats.py

-updates spacing

jordenrabasco committed:
Cleans test_q2_quality_control

-removes redundant code and cleans up spacing

jordenrabasco committed:
Cleans _stats.py and quality_control.py

-removes old code and updates spacing

jordenrabasco committed:
rounding error in histogram

-fixes rounding error in mod function in the histogram

jordenrabasco committed:
clean run_decontam.R

-removes local testing paths form file

jordenrabasco committed:
Update run_decontam.R

-removes spacing and extra comments

jordenrabasco committed:
Update error message

-updates R error messages

jordenrabasco committed:
Update citations.bib

-updates citations with decontam paper

jordenrabasco committed:
Update plugin_setup.py

-updates decontam remove description

jordenrabasco committed:
lint update: _stats, 1

-lint update

jordenrabasco committed:
lint update: _stats.py 2, _vizualizer.py 1

-lint update

jordenrabasco committed:
lint update: _vizualizer.py 2

-lint update

jordenrabasco committed:
lint update: _vizulaizer.py 3, _transformer.py 1

-lint update

jordenrabasco committed:
lint update: plugin_setup.py 1

-lint update

jordenrabasco committed:
lint update: plugin_setup.py 2

-lint update

jordenrabasco committed:
lint update: plugin_setup.py 3, quality_control.py 1

-lint update

jordenrabasco committed:
lint update: quality_control.py 2, test_quality_control.py 1

-lint update

jordenrabasco committed:
lint update: quality_control.py 3, test_quality_control.py 2

-lint update

jordenrabasco committed:
lint update: test_quality_control.py 3

-lint update

jordenrabasco committed:
lint update: test_quality_control.py 4

lint update

jordenrabasco committed:
lint update: test_quality_control.py 5, test_stats.py 1

-lint update

jordenrabasco committed:
lint update: test_stats.py 2

-lint update

jordenrabasco committed:
lint update: test_stats.py 3

-lint update

jordenrabasco committed:
lint update: setup.py 1

-lint update

jordenrabasco committed:
updates yaml file and graph

- adds R package to yaml file
- removes decrepit reference from graphing function

jordenrabasco committed:
testing error potential fix 1

-error testing in build fix

jordenrabasco committed:
testing error potential fix 2

jordenrabasco committed:
testing error potential fix 3

-move decontam tests

jordenrabasco committed:
testing error potential fix 4

jordenrabasco committed:
testing error potential fix 5

jordenrabasco committed:
test recongifuration

-reconfigures tests to come into line with CI standards

jordenrabasco committed:
lint update

jordenrabasco committed:
lint update

jordenrabasco committed:
lint update

jordenrabasco committed:
update file discovery for ci

jordenrabasco committed:
refactor and file removal

-removal of decrepit files and refactorization
-added line in setup.py to make run_decontam.R discoverable

jordenrabasco committed:
User side requested changes

-updates user api to come into line with desired changes

jordenrabasco committed:
Transpose issue fix

-fixes implementation issue associated with transposition of the ASV input matrix.
- direct fix to issue referenced in comment:
https://github.com/qiime2/q2-quality-control/pull/74#discussion_r1112298912

jordenrabasco committed:
Implementation update

-resolves all implementation comments

jordenrabasco committed:
implementation lint fix

-lint 1

jordenrabasco committed:
implementation lint 2

-lint fix 2

jordenrabasco committed:
implementation lint 3

-lint fix 3

jordenrabasco committed:
tranpose and desc format changes

-requested changes fix

jordenrabasco committed:
Control Samples error catching

-Makes sure that there are an appropriate amount of Control samples in both the ASV and metadata table

jordenrabasco committed:
removes extraneious comments

jordenrabasco committed:
clean up 1

jordenrabasco committed:
drop column update

-updates what columns to drop by being explicit and updates tests

jordenrabasco committed:
update histogram to remove NAs

-removes NA counts from histogram

jordenrabasco committed:
lint update histogram

-fixes minor lint errors

ebolyen committed:
BUG: fix ID matching

ebolyen committed:
lint

jordenrabasco committed:
tests_added_bug_fix

-changes line in run_decontam.R
-adds tests to test_decontam.py

jordenrabasco committed:
lint

jordenrabasco committed:
catch included for no metadata for samples in ASV table

-error value added

jordenrabasco committed:
Merge branch 'qiime2:master' into master

jordenrabasco committed:
Update run_decontam.R

- Updates run_decontam.R  to allow for special characters in the metadata headings

jordenrabasco committed:
initial push - hierarchical pipeline

-untested base functionality with no errors

jordenrabasco committed:
alogirthm update

jordenrabasco committed:
decontam_fast update

-intial commit fix

jordenrabasco committed:
metadata update error

-initial fix to metadata updating

jordenrabasco committed:
Update nomenclature

-updates file names
-deletes print statments

jordenrabasco committed:
optimized table generation

-Optimizes split table generation
-remove print statments

jordenrabasco committed:
further cleanup

-cleans up code

jordenrabasco committed:
initial push of updated visualizer

-initial push

jordenrabasco committed:
line update

-update when to use the line

jordenrabasco committed:
vizualizer fix

-fixes rounding error in vizualizer coloring

jordenrabasco committed:
wip

jordenrabasco committed:
Pipe update

-updated how data was being transfered

jordenrabasco committed:
lint fix _pipelines.py

jordenrabasco committed:
lint vizualizer.py

jordenrabasco committed:
lint update

jordenrabasco committed:
viz coloring fix

-fixes viz coloring with rounding issues in python

jordenrabasco committed:
lint fix

ebolyen committed:
LINT: remove parsl scraps

ebolyen committed:
lint: workspace file

ebolyen committed:
DEP: add feature-table dependency

jordenrabasco committed:
rep_seq_inital function push

-includes functionality related to the rep_seqs_trimming

jordenrabasco committed:
test first attempt

-test for rep_seqs

jordenrabasco committed:
lint updatea

jordenrabasco committed:
testing_update

- test update

jordenrabasco committed:
removal bad files

jordenrabasco committed:
Merge branch 'dev' into bad_seq_removal_expansion

jordenrabasco committed:
lint 1

linting

jordenrabasco committed:
Merge branch 'bad_seq_removal_expansion' of https://github.com/jordenrabasco/q2-quality-control into bad_seq_removal_expansion

jordenrabasco committed:
lint 2

linting

jordenrabasco committed:
lint 3

jordenrabasco committed:
minor bug error

jordenrabasco committed:
update_testing_data

-updates testing data

jordenrabasco committed:
score viz main update

-main update to vizualization both base and in the batches pipeline

jordenrabasco committed:
mild update

jordenrabasco committed:
rep-seq optional paramenter

-makes the rep seq obj optional as a parameter

jordenrabasco committed:
removal of unused style

-removes unused style

jordenrabasco committed:
label changes and styles

- updates labels and terminology

jordenrabasco committed:
small update

-small udpate

jordenrabasco committed:
lint_update 1

jordenrabasco committed:
lint 2

jordenrabasco committed:
lint 3

jordenrabasco committed:
update_score_viz

- work with new data input types for testing prep

jordenrabasco committed:
inital test and lint

jordenrabasco committed:
rudeimentary test and lint

jordenrabasco committed:
fix test

jordenrabasco committed:
removes skbio dependancy

-removes skbio dependency but keeps functionality

jordenrabasco committed:
lint update and table fix

jordenrabasco commented:
@gregcaporaso this PR is ready for review. After our conversation last week I removed the skbio dependency and added the infrastructure for testing as well a rudimentary test. I will expand the testing suite for this action when I generate the Decontam testing update PR as per our previous discussion. 
Thanks and let me know if you have any questions or concerns!

gregcaporaso commented:
Thanks @jordenrabasco! Will review asap.

gregcaporaso commented:
@jordenrabasco, this is separate from the other PR still, right? Or are they both part of this one now?

jordenrabasco commented:
> @jordenrabasco, this is separate from the other PR still, right? Or are they both part of this one now?

They should be separate unless this somehow some code snuck in while I was switching branches?

jordenrabasco commented:
> > @jordenrabasco, this is separate from the other PR still, right? Or are they both part of this one now?
> 
> They should be separate unless this somehow some code snuck in while I was switching branches?

I just looked through the files and unfortunately it looks like at some point code from the `removes contaminant sequences from representative sequence` PR got added to this one. Do you have any suggestions on how to proceed? Sorry for the confusion here and thanks!

gregcaporaso commented:
Hmm, how extensive are your edits? If they're not too extensive, it honestly might be easier to copy that code somewhere, and then do a fresh clone of this repository, re-add the changes for the two commits in two separate PRs, and I can review those. If it's a lot of code, I'll poke through and think about it more. Let me know what you think.

jordenrabasco commented:
It shouldn't be too bad, I was thinking of doing the exact same thing. 
I can ping you when I have uploaded the reversion code, if that works for you?

gregcaporaso commented:
That's perfect @jordenrabasco

jordenrabasco committed:
reversion attempt one

-removing other pull request code

jordenrabasco commented:
@gregcaporaso code in this PR should be just the score-viz update now. Everything should be in order now, let me know if you need me to change anything else for the review!

gregcaporaso commented on q2_quality_control/_pipelines.py:
```diff
@@ -8,6 +8,7 @@
 def decontam_identify_batches(ctx, table, metadata,
                               split_column,
                               method,
+                              rep_seqs=None,
```
This input doesn't actually seem to be optional based on testing on the command line:

```shell
qiime quality-control decontam-score-viz --i-decontam-scores decon_default_score_table.qza --i-table decon_default_ASV_table.qza --o-visualization viz.qzv --verbose
Traceback (most recent call last):
  File "/Users/jgc/miniconda3/envs/q2dev-amplicon/lib/python3.9/site-packages/q2cli/commands.py", line 520, in __call__
    results = self._execute_action(
  File "/Users/jgc/miniconda3/envs/q2dev-amplicon/lib/python3.9/site-packages/q2cli/commands.py", line 581, in _execute_action
    results = action(**arguments)
  File "<decorator-gen-801>", line 2, in decontam_score_viz
  File "/Users/jgc/miniconda3/envs/q2dev-amplicon/lib/python3.9/site-packages/qiime2/sdk/action.py", line 342, in bound_callable
    outputs = self._callable_executor_(
  File "/Users/jgc/miniconda3/envs/q2dev-amplicon/lib/python3.9/site-packages/qiime2/sdk/action.py", line 615, in _callable_executor_
    ret_val = self._callable(output_dir=temp_dir, **view_args)
  File "/Users/jgc/Documents/4-git-repos/qiime2/q2-quality-control/q2_quality_control/_threshold_graph/_visualizer.py", line 66, in decontam_score_viz
    for seq in rep_seqs:
TypeError: 'NoneType' object is not iterable

Plugin error from quality-control:

  'NoneType' object is not iterable

See above for debug info.
```

Can you address this and add a unit test that covers this case?

jordenrabasco replied:
A unit test in which a 'none' was passed in for the rep_seqs object was written

addressed in [973e000](https://github.com/qiime2/q2-quality-control/pull/82/commits/973e0009e469174b383036bf6faacf0f9c7dd28e)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,30 +76,125 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
+            true_dest = 'non_contam.fasta'
+            contam_dest = 'contam.fasta'
+            nan_dest = 'na_ASV_seqs.fasta'
+        with open(os.path.join(output_dir, true_dest), 'w') as fh:
```
There is a lot of duplicated code in this section that should be factored out. I have some discussion on why this is important [here](https://develop.qiime2.org/en/latest/plugins/tutorials/add-pipeline.html#update-the-nw-align-action-to-avoid-duplicating-information). Can you take another pass through and try to reduce duplicated code?

jordenrabasco replied:
This section was moved to a method in the refactoring process to remove redundant code

addressed in [8c3a5b5](https://github.com/qiime2/q2-quality-control/pull/82/commits/8c3a5b5ef158d988e3cac88141b30804175dd728)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -40,16 +41,31 @@ def _check_inputs(**kwargs):
     decontam_scores_dict = dict(decontam_scores)
 
+    # Sets rep seq flags
+    rep_seq_indicator = ["Are there rep seqs?"]
```
Can you explain what you're trying to do with this? I can try to help you figure out a better way.

jordenrabasco replied:
I am attempting to set a flag for rendering on the html side to determine when there are rep_seqs objects passed in or not.

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -40,16 +41,31 @@ def _check_inputs(**kwargs):
+    # Sets rep seq flags
+    rep_seq_indicator = ["Are there rep seqs?"]
+    temp_list = []
+    for seq in rep_seqs:
```
This is where your dependence on the optional input is coming in.

jordenrabasco replied:
A catch was implemented to allow for the optional input
 
addressed in [8c3a5b5](https://github.com/qiime2/q2-quality-control/pull/82/commits/8c3a5b5ef158d988e3cac88141b30804175dd728)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,30 +76,125 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
         table = table_dict[key]
         decontam_scores = decontam_scores_dict[key]
-        df = decontam_scores.to_dataframe()
+        df = decontam_scores
```
Why rename this? I think it would help with interpretability if you kept referring to it as `decontam_scores`.

jordenrabasco replied:
This was removed as suggested and the name was kept at decontam_scores

addressed in [8c3a5b5](https://github.com/qiime2/q2-quality-control/pull/82/commits/8c3a5b5ef158d988e3cac88141b30804175dd728)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,30 +76,125 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
         filt_read_nums = read_nums[p_vals.index]
 
         contams = (p_vals < threshold)
+        contam_indices = []
```
This section would benefit from some comments, and I think a lot of this could be refactored to pandas commands (e.g., you can slice based on boolean arrays, which seems to be what you're doing here). Can you take another pass through this and try to simplify and clarify it?

jordenrabasco replied:
This section was refactored as suggested

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

gregcaporaso reviewed (changes requested):
@jordenrabasco, I'm out of time for reviewing right now, but I think this still needs a bit of work. Can you take a pass through my comments and let me know if you have questions. It will also help if you can link or attach example input data and output data to the pull request, to help with testing and understanding what the output should look like.

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -40,16 +41,31 @@ def _check_inputs(**kwargs):
 
-def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
-                       table: pd.DataFrame, threshold: float = 0.1,
+def format_fasta(header, sequence):
```
Don't reimplement this, as you're going to need to write unit tests for it, handle edge cases, etc. 

I think a better way to do this is to bring your `rep_seqs` in as a DNAInterator (see an example [here](https://github.com/caporaso-lab/q2-dwq2/pull/12/files#diff-12dca64616ef2e38c0f6ad05c6050f649209daab810693d78a3ece281d0effbe)). Then, you can iterate over the items to write them to file - for example:

```python
with open('test.fasta', 'w') as fh:
    for seq in rep_seqs:
        seq.write(fh)
```

jordenrabasco replied:
This was removed in favor of the data being read in as a DNAIterator as suggested

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

jordenrabasco committed:
inital comment update

-adds comments
-reduces redunant code
-fixes bug in optional rep-seq obj with score-viz function

jordenrabasco committed:
minor rep_seq  code update

-moved like code together for readability

jordenrabasco committed:
testing update

-adds test for no rep_seqs
-adds readability spacing to vizualizer.py

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -144,12 +147,12 @@ def decontam_score_viz(output_dir, decontam_scores: pd.DataFrame,
             contam_rep_seqs = ["NNNNNNNNNNNNN"] * len(contam_indices)
 
+
+        #legacy code splitting na and true asvs kept here for future iterations of plugin
```
I'm not sure what you mean by legacy code, but if you're not using it now you should drop it. you can always re-access code from your commit history.

jordenrabasco replied:
legacy code sections are dropped

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -41,15 +43,48 @@ def _check_inputs(**kwargs):
+                           "=nt&CMD=Put&QUERY=%s")
+    with open(os.path.join(output_dir, dest), 'w') as fh:
+        for index, row in df.iterrows():
+            fh.write(format_fasta(row['id'], row['Sequence']))
```
See my earlier comment about not implementing this yourself, or write unit tests for `format_fasta`

jordenrabasco replied:
format_fasta function removed in favor  of the data being read in as a DNAIterator

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,31 +95,112 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
+                contam_indices)]['Sequence'].tolist()
+            true_rep_seqs = rep_seqs_df[rep_seqs_df.index.isin(
+                true_indices)]['Sequence'].tolist()
+        else:
```
Can you avoid this somehow?

jordenrabasco replied:
removed this code in refactoring process

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,31 +95,112 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
     gray_lab_arr = []
     red_lab_arr = []
     blue_lab_arr = []
+    data_arr = []
```
What are each of these?

jordenrabasco replied:
comments added and depreciated arrays were removed

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -41,15 +43,48 @@ def _check_inputs(**kwargs):
+                       threshold: float = 0.1,
                        weighted: bool = True, bin_size: float = 0.02):
+
+    #checks inputs
```
What is this doing? Is it needed? QIIME 2 should be checking the inputs for you already.

jordenrabasco replied:
This is checking to make that the inputs are appropriate. There is crossover with what Qiime2 checks however it also checking that the threshold  value is between 0 and 1. Should this be checked on the Qiime2 framework side of things? Or should I just raise a valueerror if it is out of bounds? Thanks!

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -60,31 +95,112 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
+        true_indices = []
+
+        #parses index for contaminant identification
+        for true_or_false_index in contams.index:
```
See my previous comment - you shouldn't need to do all of this.

jordenrabasco replied:
This was removed in favor of dataframe slicing

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)

jordenrabasco committed:
refactoring, comments, removal of deprecated code

- refactors code and changes the rep seq read in to DANIterator
- adds numerous comments explaining various features of the code
-removes old code no longer pertainent to the plugin

jordenrabasco commented:
Attached is a folder with example data and output as requested.
Folder with data: [decontam_score_viz_testing.zip](https://github.com/qiime2/q2-quality-control/files/15416786/decontam_score_viz_testing.zip)

The commands that I used for each of the outputs are as follows;

batches rep-seq:

qiime quality-control decontam-identify-batches --i-table table.qza --i-rep-seqs rep_seqs.qza --m-metadata-file metadata.tsv --p-split-column subject --p-filter-empty-features  --p-method combined --p-prev-control-column Sample_or_Control --p-prev-control-indicator control --p-freq-concentration-column Concentration --p-threshold 0.1 --p-no-weighted --p-bin-size 0.05 --output-dir rep_seqs_batches_outputs

batches no-rep-seq:

qiime quality-control decontam-identify-batches --i-table table.qza  --m-metadata-file metadata.tsv --p-split-column subject --p-filter-empty-features  --p-method combined --p-prev-control-column Sample_or_Control --p-prev-control-indicator control --p-freq-concentration-column Concentration --p-threshold 0.1 --p-no-weighted --p-bin-size 0.05 --output-dir no_rep_seqs_batches_outputs

base decontam score viz rep-seq:

qiime quality-control decontam-score-viz --i-decontam-scores comb_decontam_scores.qza --i-table table.qza --i-rep-seqs rep_seqs.qza --p-threshold 0.1 --p-no-weighted --p-bin-size 0.05 --o-visualization rep_seqs_decontam_score_viz.qzv

base decontam score viz no_rep_seq:

qiime quality-control decontam-score-viz --i-decontam-scores comb_decontam_scores.qza --i-table table.qza  --p-threshold 0.1 --p-no-weighted --p-bin-size 0.05 --o-visualization no_rep_seqs_decontam_score_viz.qzv

Additionally for completeness sake included a link to a tutorial I have been working on where you can download the raw files and outputs with explanations.  You should be able to find it [here](https://github.com/jordenrabasco/q2-decontam-tutorial?tab=readme-ov-file)

Let me know if you have any issues with this, and thanks so much!

jordenrabasco committed:
if statement and comment cleanup

jordenrabasco committed:
testing update and cleanup

-updates testing to bring into line with DNA iterator
-includes dependency on skbio.sequence

jordenrabasco committed:
comment update

-add q2-dada2 source comment sections

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -27,9 +28,11 @@
 }
 
 
+#checks inputs
```
This is no longer needed now that we're using `Range` to ensure that these values are all in the right range. We removed this.

gregcaporaso committed:
replaced check inputs with Range on parameters

gregcaporaso committed:
replaced check inputs with Range on parameters

gregcaporaso committed:
replace test list logic

gregcaporaso committed:
resolve merge conflicts with qiime2:dev

gregcaporaso reviewed (changes requested):
Pair-reviewed with @jordenrabasco. 

@jordenrabasco is going to replace the `rep_seq_indicator` variable list with a boolean value, and get flake8 to pass. This is then good to merge. Some remaining issues will be addressed as part of #86.

jordenrabasco committed:
rep_seq_indicator update and testing update

jordenrabasco committed:
lint fix one

jordenrabasco committed:
lint 2

jordenrabasco committed:
lint3

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
-    gray_lab_arr = []
-    red_lab_arr = []
-    blue_lab_arr = []
+    # TODO: Replace rep_seq_indicator to a boolean value
+    # Sets rep seq flags if rep_seq indciator is >1 then
+    # the sequences are printed in table otherwise no sequences are
+    # printed and temp lists
```
```suggestion
    # Indicates whether sequences are provided, and therefore 
    # whether a section should be created for them in the viz
```

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
+    # Sets rep seq flags if rep_seq indciator is >1 then
+    # the sequences are printed in table otherwise no sequences are
+    # printed and temp lists
+    rep_seq_indicator = True
```
```suggestion
    rep_seq_indicator = rep_seqs is not None
```

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
+        # objects are inalized for true seqe, NA seqs, and contaminant seqs
+        contam_rep_seqs = []
+        true_rep_seqs = []
+        if rep_seqs is not None:
```
```suggestion
        if rep_seq_indicator:
```

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
+        nan_indices = decontam_scores[
+            decontam_scores['p'].isna()].index.tolist()
+
+        # if rep reqs are not found then dummy variables
```
I think this bit about dummy variables is no longer relevant

jordenrabasco replied:
removed and added a bit more about the logic of the replacement code 

addressed in [b466ed9](https://github.com/qiime2/q2-quality-control/pull/82/commits/b466ed97b7fe9af561c1e13576a6c52a19c79fa6)

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
+                    # TODO:
+                    #  describe the case where you'd end up here in a comment
+                    pass
+        else:
+            rep_seq_indicator = False
```
```suggestion
```

gregcaporaso commented on q2_quality_control/_threshold_graph/_visualizer.py:
```diff
@@ -7,84 +7,169 @@
+                    true_rep_seqs.append(seq)
+                else:
+                    # TODO:
+                    #  describe the case where you'd end up here in a comment
```
Can you fill a comment in here?

jordenrabasco replied:
Added comments describing the case

addressed in [b466ed9](https://github.com/qiime2/q2-quality-control/pull/82/commits/b466ed97b7fe9af561c1e13576a6c52a19c79fa6)

jordenrabasco committed:
Update q2_quality_control/_threshold_graph/_visualizer.py

Co-authored-by: Greg Caporaso <192372+gregcaporaso@users.noreply.github.com>

jordenrabasco committed:
Update q2_quality_control/_threshold_graph/_visualizer.py

Co-authored-by: Greg Caporaso <192372+gregcaporaso@users.noreply.github.com>

jordenrabasco committed:
Update q2_quality_control/_threshold_graph/_visualizer.py

Co-authored-by: Greg Caporaso <192372+gregcaporaso@users.noreply.github.com>

jordenrabasco committed:
Update q2_quality_control/_threshold_graph/_visualizer.py

Co-authored-by: Greg Caporaso <192372+gregcaporaso@users.noreply.github.com>

jordenrabasco committed:
clean up comments

-comments cleaned up and spacing

jordenrabasco committed:
Merge branch 'score_viz_update' of https://github.com/jordenrabasco/q2-quality-control into score_viz_update

jordenrabasco committed:
lint4
